src/libraries/System.Net.Http/tests/FunctionalTests/ResponseStreamConformanceTests.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.IO.Tests;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Http.Functional.Tests
+{
+    public sealed class Http1CloseResponseStreamConformanceTests : ResponseConnectedStreamConformanceTests
+    {
+        protected override string GetResponseHeaders() => "HTTP/1.1 200 OK\r\nConnection: close\r\n\r\n";
+
+        protected override async Task<StreamPair> CreateConnectedStreamsAsync()
+        {
+            StreamPair pair = await base.CreateConnectedStreamsAsync();
+            Assert.False(pair.Stream2.CanWrite);
+            Assert.True(pair.Stream2.CanRead);
+            return pair;
+        }
+    }
+
+    public sealed class Http1RawResponseStreamConformanceTests : ResponseConnectedStreamConformanceTests
+    {
+        protected override string GetResponseHeaders() => "HTTP/1.1 101 Switching Protocols\r\n\r\n";
+
+        protected override async Task<StreamPair> CreateConnectedStreamsAsync()
+        {
+            StreamPair pair = await base.CreateConnectedStreamsAsync();
+            Assert.True(pair.Stream2.CanWrite);
+            Assert.True(pair.Stream2.CanRead);
+            return pair;
+        }
+    }
+
+    public sealed class Http1ContentLengthResponseStreamConformanceTests : ResponseStandaloneStreamConformanceTests
+    {
+        protected override async Task WriteResponseAsync(Stream responseStream, byte[] bodyData)
+        {
+            await responseStream.WriteAsync(Encoding.ASCII.GetBytes($"HTTP/1.1 200 OK\r\nContent-Length: {bodyData.Length}\r\n\r\n"));
+            await responseStream.WriteAsync(bodyData);
+        }
+    }
+
+    public sealed class Http1ChunkedResponseStreamConformanceTests : ResponseStandaloneStreamConformanceTests
+    {
+        protected override async Task WriteResponseAsync(Stream responseStream, byte[] bodyData)
+        {
+            await responseStream.WriteAsync(Encoding.ASCII.GetBytes($"HTTP/1.1 200 OK\r\nTransfer-Encoding: chunked\r\n\r\n{bodyData.Length:X}\r\n"));
+            await responseStream.WriteAsync(bodyData);
+            await responseStream.WriteAsync(Encoding.ASCII.GetBytes("\r\n"));
```

Comment:
It would be nice to have more variations here. E.g. byte per chunk. Doesn't have to happen as part of this PR of course.

Reply:
I'll leave adding dribble support to you :)

Reply:
Dribble is different -- dribble applies across all response types, and is really about fragmenting bytes on the wire without changing what's actually sent.

Byte per chunk is specifically about chunked encoding and changes what's sent on the wire.

Reply:
Ah, I misunderstood what you were suggesting. Sure, I can easily add a byte-per-chunk variant in this PR if you think it's valuable.

Reply:
I do think it's valuable, yeah

Reply:
We had/have tests like this in kestrel to test the parser logic. It's great to do but can also be slow (I think we delayed between every byte)

Reply:
Yeah, there are such tests elsewhere in HttpClient tests as well, e.g. things that use https://github.com/dotnet/runtime/blob/b25c7c47410cb43d33fbacc6da7f32bba0d1f00b/src/libraries/Common/tests/System/Net/Http/LoopbackServer.cs#L356.  And separately ones that dribble the content with delays, via https://github.com/dotnet/runtime/blob/b25c7c47410cb43d33fbacc6da7f32bba0d1f00b/src/libraries/Common/tests/System/Net/Http/DribbleStream.cs.